### PR TITLE
Implement global shortcuts

### PR DESCRIPTION
**Pull request by benebsiny**
See https://github.com/asticode/go-astilectron/pull/393

### CODE REVIEW

**Comment by benebsiny** [comment]
Off topic, won't it be a good idea to add tabs to the statements inside the `case`?

File: src/consts.js [hunk 1]
```diff
@@ -26,6 +26,12 @@ module.exports = {
         dockEventHidden: "dock.event.hidden",
         dockEventIconSet: "dock.event.icon.set",
         dockEventShown: "dock.event.shown",
+        globalShortcutCmdRegister: "global.shortcut.cmd.register",
```
Comment (asticode) [inline src/consts.js]: Could you make sure there's an `s` at the end of all `shortcut` occurrences?

File: src/consts.js [hunk 2]
```diff
@@ -26,6 +26,12 @@ module.exports = {
...
+        globalShortcutCmdIsRegistered: "global.shortcut.cmd.is.register",
+        globalShortcutCmdUnregister: "global.shortcut.cmd.unregister",
+        globalShortcutCmdUnregisterAll: "global.shortcut.cmd.unregister.all",
+        globalShortcutEventProcessFinished: "global.shortcut.event.process.finished",
```
Comment (asticode) [inline src/consts.js]: Could you remove this event and instead use:

- `global.shortcuts.event.registered`
- `global.shortcuts.event.is.registered`
- `global.shortcuts.event.unregistered`
- `global.shortcuts.event.unregistered.all`

File: index.js [hunk 3]
```diff
@@ -349,6 +349,26 @@ function onReady () {
...
+
+            // Global Shortcut
+            case consts.eventNames.globalShortcutCmdRegister:
+            const isRegistered = globalShortcut.register(json.globalShortcut.accelerator, () => {
+                client.write(json.targetID, consts.eventNames.globalShortcutEventTriggered, {GlobalShortcut:{accelerator: json.globalShortcut.accelerator}});
+            });
+            client.write(json.targetID, consts.eventNames.globalShortcutEventProcessFinished, {GlobalShortcut:{isRegistered: isRegistered}});
+            break;
+            case consts.eventNames.globalShortcutCmdIsRegistered:
+            const isRegistered2 = globalShortcut.isRegistered(json.globalShortcut.accelerator);
```
Comment (asticode) [inline index.js]: No need to create a constant here, you can directly call:

```js
client.write(json.targetID, consts.eventNames.globalShortcutsEventIsRegistered, {globalShortcut:{isRegistered: globalShortcut.isRegistered(json.globalShortcut.accelerator)}});
```

File: index.js [hunk 4]
```diff
@@ -349,6 +349,26 @@ function onReady () {
             case consts.eventNames.windowCmdWebContentsExecuteJavascript:
             elements[json.targetID].webContents.executeJavaScript(json.code).then(() => client.write(json.targetID, consts.eventNames.windowEventWebContentsExecutedJavaScript));
             break;
+
+            // Global Shortcut
+            case consts.eventNames.globalShortcutCmdRegister:
+            const isRegistered = globalShortcut.register(json.globalShortcut.accelerator, () => {
+                client.write(json.targetID, consts.eventNames.globalShortcutEventTriggered, {GlobalShortcut:{accelerator: json.globalShortcut.accelerator}});
```
Comment (asticode) [inline index.js]: You need to use `globalShortcuts` instead of `GlobalShortcut` in the json

**Review by asticode — CHANGES_REQUESTED** [review]

File: src/consts.js [hunk 5]
```diff
@@ -26,12 +26,12 @@ module.exports = {
...
-        globalShortcutCmdUnregister: "global.shortcut.cmd.unregister",
-        globalShortcutCmdUnregisterAll: "global.shortcut.cmd.unregister.all",
-        globalShortcutEventProcessFinished: "global.shortcut.event.process.finished",
-        globalShortcutEventTriggered: "global.shortcut.event.triggered",
+        globalShortcutEventRegistered: "global.shortcuts.event.registered",
+        globalShortcutEventIsRegistered: "global.shortcuts.event.is.registered",
+        globalShortcutEventUnregistered: "global.shortcuts.event.unregistered",
+        globalShortcutEventUnregisteredAll: "global.shortcuts.event.unregistered.all",
+        globalShortcutEventProcessFinished: "global.shortcuts.event.process.finished",
+        globalShortcutEventTriggered: "global.shortcuts.event.triggered",
```
Comment (asticode) [inline src/consts.js]: We misunderstood each other: could you rollback all `globalShortcutCmd...` events but replace the `globalShortcutEventProcessFinished` with:

- global.shortcuts.event.
- global.shortcuts.event.is.
- global.shortcuts.event.
- global.shortcuts.event.unregistered.all

Also, could you make sure there's an `s` at the end of `globalShortcut` in the key?

This should look like this:

```javascript
globalShortcutsCmdRegister: "global.shortcuts.cmd.register",
globalShortcutsCmdIsRegistered: "global.shortcuts.cmd.is.register",
globalShortcutsCmdUnregister: "global.shortcuts.cmd.unregister",
globalShortcutsCmdUnregisterAll: "global.shortcuts.cmd.unregister.all",
globalShortcutsEventRegistered: "global.shortcuts.event.registered",
globalShortcutsEventIsRegistered: "global.shortcuts.event.is.registered",
globalShortcutsEventUnregistered: "global.shortcuts.event.unregistered",
globalShortcutsEventUnregisteredAll: "global.shortcuts.event.unregistered.all",
globalShortcutsEventProcessFinished: "global.shortcuts.event.process.finished",
globalShortcutsEventTriggered: "global.shortcuts.event.triggered",
```

`globalShortcutsCmd...` are the events received by javascript, `globalShortcutsEvent...` are the events sent by javascript.

File: index.js [hunk 6]
```diff
@@ -351,21 +351,20 @@ function onReady () {
             break;
 
             // Global Shortcut
-            case consts.eventNames.globalShortcutCmdRegister:
-            const isRegistered = globalShortcut.register(json.globalShortcut.accelerator, () => {
-                client.write(json.targetID, consts.eventNames.globalShortcutEventTriggered, {GlobalShortcut:{accelerator: json.globalShortcut.accelerator}});
+            case consts.eventNames.globalShortcutEventRegistered:
+            const isRegistered = globalShortcut.register(json.globalShortcuts.accelerator, () => {
+                client.write(json.targetID, consts.eventNames.globalShortcutEventTriggered, {GlobalShortcuts:{accelerator: json.globalShortcuts.accelerator}});
```
Comment (asticode) [inline index.js]: You need to use `globalShortcuts` with a lower `g` instead of `GlobalShortcuts` in the json. Same thing below

**Review by asticode — CHANGES_REQUESTED** [review]
Ping me when changes have been made 👍

**Comment by asticode** [comment]
👍 I'll merge this PR once its GO counter-part is OK

> Off topic, won't it be a good idea to add tabs to the statements inside the case?

If you feel like it, that would be a good idea indeed 👍

**Comment by asticode** [comment]
Thanks for the PR! ❤️

**Pull request merged**